src/WaCore.Data.Ef/WacEfUnitOfWork.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WaCore.Contracts.Data;
+
+namespace WaCore.Data.Ef
+{
+    public class WacEfUnitOfWork<TDbContext> : IWacUnitOfWork
+        where TDbContext : DbContext, IWacDbContext
+    {
+        protected readonly TDbContext _dbContext;
+        protected readonly IServiceProvider _serviceProvider;
+        protected Dictionary<Type, object> _repositories = new Dictionary<Type, object>();
+
+        public TDbContext DbContext => _dbContext;
+
+
+        public WacEfUnitOfWork(TDbContext dbContext, IServiceProvider serviceProvider)
+        {
+            _dbContext = dbContext;
+            _serviceProvider = serviceProvider;
+        }
+
+        public async Task<IWacTransaction> BeginTransactionAsync(IsolationLevel isolationLevel, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return new WacDbContextTransactionWrapper(await DbContext.Database.BeginTransactionAsync(isolationLevel, cancellationToken));
+        }
+
+        public async Task<IWacTransaction> BeginTransactionAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return new WacDbContextTransactionWrapper(await DbContext.Database.BeginTransactionAsync(cancellationToken));
+        }
+
+        public IWacTransaction BeginTransaction(IsolationLevel isolationLevel)
+        {
+            return new WacDbContextTransactionWrapper(DbContext.Database.BeginTransaction(isolationLevel));
```

Comment:
One thing to discuss. What will be if user opens two transactions? Should we give him new transaction object or should we check whether there is already active transaction?